### PR TITLE
Add package:xfce4-appfinder

### PR DESCRIPTION
**Pull request by suhan-paradkar**
Will be a draft until intltool gets merged

### CODE REVIEW

**Comment by Yisus7u7** [comment]
One question, as you do to build xfce4 plug-ins without using Xfce4-dev-tools?, why when I try to build an xfce4 plugin you ask me to install Xfce4-dev-tools, otherwise it doesn't work

**Comment by suhan-paradkar** [comment]
@Yisus7u7 xfce4-appfinder does not require xfce4-dev-tools...
I have successfully cross-compiled xfce4-appfinder in the build environment...
The only thing left is inclusion of Intltool
Although, some xfce panel plugins require it as a dependency...
xfce4-dev-tools has a few problems with cross-compilation as of now...

**Comment by Yisus7u7** [comment]
Okay, I tried to compile Xfce4-dev-tools, as the program the compilation was successful, but I didn't see anywhere the binaries

**Comment by WMCB-Tech** [comment]
xfce4-dev-tools dosen't have binaries

As stated;
> The Xfce development tools are a collection of tools and macros for Xfce developers and people that want to build Xfce from git. In addition, it contains the Xfce developer's handbook.

So it only contains some macros and libs

**Comment by WMCB-Tech** [comment]
Probably you forgot to `make install`?

**Comment by Yisus7u7** [comment]
A ok, but still compile it, edit the paths and the make install was successful, but I still did not have xfce4-dev-tools installed

**Comment by WMCB-Tech** [comment]
Check if you have one of the following on your `$PREFIX/bin`

```
xdt-autogen
xdt-csource
xfce-build
xfce-do-release
xfce-get-release-notes
xfce-get-translations
xfce-update-news
```
I'm pretty sure that's included in `xfce4-dev-tools`

**Comment by suhan-paradkar** [comment]
> Check if you have one of the following on your `$PREFIX/bin`
> 
> ```
> xdt-autogen
> xdt-csource
> xfce-build
> xfce-do-release
> xfce-get-release-notes
> xfce-get-translations
> xfce-update-news
> ```
> 
> I'm pretty sure that's included in `xfce4-dev-tools`

You are right...
As of now, xfce4-dev-tools on device compilation succeeds.. but cross compilation stops at xdt-autogen

**Comment by Yisus7u7** [comment]
Here's some information about when I've been compiling it:

https://gitlab.xfce.org/xfce/xfce4-dev-tools/-/issues/48#note_31475

I'll try again

**Comment by Yisus7u7** [comment]
Screenshot_20210513-093153

I had to use termux in chroot mode
Is that what you're looking for?

**Comment by Yisus7u7** [comment]
Screenshot_20210513-093509

**Comment by Yisus7u7** [comment]
@suhan-paradkar?

**Comment by suhan-paradkar** [comment]
No... The cross compilation is done in a docker container using ./build-package.sh located in termux-packages/scripts

**Comment by Yisus7u7** [comment]
I mean, if that container isn't generated, it won't work?

**Comment by suhan-paradkar** [comment]
> I mean, if that container isn't generated, it won't work?

You can't cross compile for other architectues without the container.. In ondevice build, ./build-package.sh builds for only that device architecture

**Comment by Yisus7u7** [comment]
Okay, that means I can create a deb package for my architecture and its will do?

**Comment by suhan-paradkar** [comment]
> Okay, that means I can create a deb package for my architecture and its will do?

Yes... But only on your device..

**Comment by Yisus7u7** [comment]
Screenshot_20210513-102131
@suhan-paradkar Build Xfce4-dev-tools in a package for the arm architecture, package, and its man pages, all I need is intltool.

**Comment by Yisus7u7** [comment]
Screenshot_20210513-102502

There are problems with construction intltool right?

File: packages/xfce4-appfinder/build.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+TERMUX_PKG_HOMEPAGE=https://docs.xfce.org/xfce/xfce4-appfinder/start
+TERMUX_PKG_DESCRIPTION="Application launcher and finder"
+TERMUX_PKG_LICENSE="GPL-2.0"
+_MAJOR_VERSION=4.16
+TERMUX_PKG_VERSION=$_MAJOR_VERSION.1
+TERMUX_PKG_SRCURL=https://archive.xfce.org/src/xfce/xfce4-appfinder/$_MAJOR_VERSION/xfce4-appfinder-$TERMUX_PKG_VERSION.tar.bz2
+TERMUX_PKG_SHA256=bfe3e9bd92695014ee74a2fbb7f5fd1b4c29cf043c4a11598b8958324c81e7ec
+TERMUX_PKG_DEPENDS="intltool, glib, gtk3, libxfce4ui, garcon"
```
Comment (xeffyr) [inline packages/xfce4-appfinder/build.sh:8]: Does it really has `intltool` uses runtime dependency? If not - remove it.

P.S. https://github.com/termux/termux-packages/blob/master/scripts/setup-ubuntu.sh#L35

**Comment by Yisus7u7** [comment]
@xeffyr 
Yes, apparently intltool is a necessary dependency for Xfce4-dev-tools, because when I try to compile xfce4-notes-plugin, if I detected xfce4-dev-tools but then told me I should have "intltool" installed to continue

**Comment by xeffyr** [comment]
> Yes, apparently intltool is a necessary dependency for Xfce4-dev-tools

This pull request about xfce4-appfinder, isn't it?

File: packages/xfce4-appfinder/build.sh (hunk 1 above)
Comment (suhan-paradkar) [reply packages/xfce4-appfinder/build.sh:8]: Yes.., apprently, it searches for intltool using pkg-config during step configure

**Comment by suhan-paradkar** [comment]
> > Yes, apparently intltool is a necessary dependency for Xfce4-dev-tools
> 
> This pull request about xfce4-appfinder, isn't it?

Yes it is

**Comment by Yisus7u7** [comment]
And then how do I install intltool on my device?

**Comment by Yisus7u7** [comment]
@xeffyr
intltool is not required as a dependency, but if as a recommended package, to compile xfce4-dev-tools you do not need intltool, but xfce4 plugins ask for it when compiling it, therefore intltool is a necessary package, not for compilation but for proper operation

**Comment by Yisus7u7** [comment]
@xeffyr Could you add intltool?

**Comment by Grimler91** [comment]
> And then how do I install intltool on my device?

intltool is currently not an available package, you need to build it from source to get it currently.

> Could you add intltool?

Intltool requires perl modules though, and perl modules are truly annoying when it comes to cross-compilation.
Nevertheless, there is a PR to add it: https://github.com/termux/termux-packages/pull/6755. 
Due to the perl modules dependency it should be tested to make sure it works in a fresh termux installation before merging.

> intltool is not required as a dependency, but if as a recommended package, to compile xfce4-dev-tools you do not need intltool, but xfce4 plugins ask for it when compiling it, therefore intltool is a necessary package, not for compilation but for proper operation

Intltool is a build (opposite of runtime) dependency for xfce4-appfinder, which is the package this PR adds.
Since it is a build time dependency it should not be specified as TERMUX_PKG_DEPENDS. And since intltool is installed from the ubuntu repos in the docker image rather than through termux it should not be specified in TERMUX_PKG_BUILD_DEPENDS either.

**Comment by Yisus7u7** [comment]
Oh ok, then I'll wait for one day to be added. 
Pd : I don't have pc

**Comment by suhan-paradkar** [comment]
> Since it is a build time dependency it should not be specified as TERMUX_PKG_DEPENDS. And since intltool is installed from the ubuntu repos in the docker image rather than through termux it should not be specified in TERMUX_PKG_BUILD_DEPENDS either.

I'll change it

**Pull request closed without merging**